### PR TITLE
Feature describe config options

### PR DESCRIPTION
**Pull request by dmolineus**
See #17

### CODE REVIEW

**Comment by discordier** [comment]
The tests will fail until I adjust the bootstraps, however, as soon as I do that, all other implementations will start to fail. How shall we tackle this?

**Comment by dmolineus** [comment]
> Who is going to perform the validation now?
Shall we move it to to command or dedicate a processor class to it?

Right now it's called by the command. Later on I prefer a processor class which can be reused in the validate command

**Comment by dmolineus** [comment]
> The tests will fail until I adjust the bootstraps, however, as soon as I do that, all other implementations will start to fail. How shall we tackle this?

I suggest using a feature branch on the repository builder which applies the changes and would build the inline repository of phpcq.

File: src/Plugin/Config/PhpcqConfigOptionsBuilder.php [hunk 1]
```diff
@@ -15,7 +15,7 @@ final class PhpcqConfigOptionsBuilder implements ConfigOptionsBuilderInterface
     public function describeArrayOption(
         string $name,
         string $description,
-        array $defaultValue,
+        array $defaultValue = [],
```
Comment (discordier) [inline src/Plugin/Config/PhpcqConfigOptionsBuilder.php]: This makes it impossible to have no default at all. I think we should make this nullable as well, as an empty array is not always correct.

Comment (dmolineus) [reply src/Plugin/Config/PhpcqConfigOptionsBuilder.php]: Okay, I changed it. I couldn't imagine a case there it would be needed

**Comment by dmolineus** [comment]
From my point of view this PR is ready and could be merged. I updated the local repository and tests are passing again.

**Comment by dmolineus** [comment]
I extracted the `validateConfig` method into a new `ConfigOptions` class as it doesn't belong to a builder.

**Pull request merged**